File: javascript/examples/smoke.mjs
import { CseApiDocsDeepenClient } from "../client.mjs";

const client = new CseApiDocsDeepenClient({ defaultDelayMs: 1000 });
console.log("smoke", CseApiDocsDeepenClient.slug, "->", "topGainers");
const data = await client.topGainers();
const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
console.log("ok", preview);

File: python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from cse_api_docs_deepen import CseApiDocsDeepenClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with CseApiDocsDeepenClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "top_gainers")
    data = client.top_gainers()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)
